questions/currying.md

```diff
@@ -0,0 +1,63 @@
+### How do you get both statements to evaluate to the same thing?
+
+```
+console.log(sum(2,3));   // Outputs 5
+console.log(sum(2)(3));  // Outputs 5
+```
+
+#### Answer
+
+This is an example of Currying, which is the process of breaking down a function into a series of functions that each take a single argument.
+
+In simplest terms, the second expression can be expressed as follows:
+
+```
+function sum (a) {
+  return function (b) {
+    return a + b;
+  }
+}
+
+sum(2)(3) // returns 5
+```
+
+In this example, the argument `a` of `sum(2)` is accessible by the inner `function (b)`, and therefore can perform the operation `a + b`, and pass the response back.
+
+Another way to visualize this would be:
+
+```
+const sum = (a) => (b) => a + b;
+```
+
+As you can see, we're passing one function into another to evaluate an expression.
+
+However, this will not handle the first case of `sum(2,3)` unless we handle the potential second argument.
+
+In this case, we should use a slight conditional to handle both cases.
+
+```
+function sum (a, b) {
+  const curry = (b) => a + b;
+  return b ? curry(b) : curry;
```

Comment:
Also just realized this will fail on 0. We should check if `b !== undefined`

Reply:
Sangat baik